app/models.py

```diff
@@ -2332,27 +2332,33 @@ class BroadcastEvent(db.Model):
     transmitted_starts_at = db.Column(db.DateTime, nullable=True)
     transmitted_finishes_at = db.Column(db.DateTime, nullable=True)
 
-    # @property
-    # def reference(self):
-    #     # TODO: write this `from_event` function
-    #     return BroadcastMessageTemplate.from_event(self.serialize()).reference
+    @property
+    def reference(self):
+        return BroadcastMessageTemplate.from_event(self.serialize()).reference
+
+    def get_earlier_message_references(self):
+        from app.dao.broadcast_message_dao import get_earlier_events_for_broadcast_event
+        return [event.reference for event in get_earlier_events_for_broadcast_event(self.id)]
 
     def serialize(self):
         return {
-            'id': self.id,
+            'id': str(self.id),
 
-            'service_id': self.service_id,
+            'service_id': str(self.service_id),
 
-            # 'reference': self.reference,
```

Comment:
I'm not sure what is going on here: I get that we add `previous_event_references` to be able to get all related events into one bundle, but why are we deleting the single `reference`?

Reply:
so didnt realise it when i first wrote the function but `self.reference` is a property, that gets the reference by calling `BroadcastMessageTemplate.from_event(self.serialize()).reference`. so serialize would call `self.reference` which would call `self.serialize` which would call `self.reference`  .... meant an infinite recursion.